.travis.yml

```diff
@@ -22,3 +21,60 @@ notifications:
     on_success: change
     on_failure: always
     on_start: never
+env:
+  global:
+    - DISPLAY=:99.0
+  matrix:
+    - TARGET=phantom
+    - TARGET=chrome
+    - TARGET=firefox
+matrix:
+  allow_failures:
+    - env: TARGET=chrome
+    - env: TARGET=firefox
+  fast_finish: true
+
+before_script:
+  - sh -e /etc/init.d/xvfb start
+  - sleep 3 # give xvfb some time to start
+  - echo "MAVEN_OPTS='-Xmx1g -Xms256m'" > ~/.mavenrc
+script:
+  - |
+      if [[ $TARGET == *"phantom"* ]]; then
+        sh -e utilities/travis-run-ci-tests.sh
+        if [[ $? == 0 ]]; then
+          sh -e utilities/travis-push-javadoc-to-gh-pages.sh
+          sh -e utilities/travis-deploy-to-sonatype.sh
+        fi
+      fi
+  - |
+      if [[ $TARGET == *"chrome"* ]]; then
+        sudo apt-get -y purge chromium-browser
+        export CHROME_REVISION=`curl -s http://commondatastorage.googleapis.com/chromium-browser-snapshots/Linux_x64/LAST_CHANGE`
+        curl -L -O "http://commondatastorage.googleapis.com/chromium-browser-snapshots/Linux_x64/${CHROME_REVISION}/chrome-linux.zip"
+        unzip chrome-linux.zip
+        # See https://sites.google.com/a/chromium.org/chromedriver/capabilities#TOC-Using-a-Chrome-executable-in-a-non-standard-location
+        sudo ln -sf $PWD/chrome-linux/chrome-wrapper /usr/bin/google-chrome
+        export CHROMEDRIVER_VERSION=`curl -s http://chromedriver.storage.googleapis.com/LATEST_RELEASE`
+        curl -L -O "http://chromedriver.storage.googleapis.com/${CHROMEDRIVER_VERSION}/chromedriver_linux64.zip"
+        unzip chromedriver_linux64.zip && chmod +x chromedriver && sudo mv chromedriver /usr/local/bin
+        google-chrome --version
+        mvn test -pl client -DsuiteXmlFile=Chrome-Suite.xml -DSELION_SELENIUM_RUN_LOCALLY=true \
+          -DSELION_SELENIUM_CHROMEDRIVER_PATH=/usr/local/bin/chromedriver \
+          -DSELION_SELENIUM_CHROME_OPTIONS="--no-sandbox" -B -V
+      fi
```

Comment:
Is this approach better than running against Sauce? It seems to come at a cost..